src/plugins/wizard/public/visualizations/common/expression_helpers.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { cloneDeep } from 'lodash';
+import { OpenSearchaggsExpressionFunctionDefinition } from '../../../../data/public';
+import { ExpressionFunctionOpenSearchDashboards } from '../../../../expressions';
+import { buildExpressionFunction } from '../../../../expressions/public';
+import { VisualizationState } from '../../application/utils/state_management';
+import { getAggService, getIndexPatterns } from '../../plugin_services';
+
+export const getAggExpressionFunctions = async (visualization: VisualizationState) => {
+  const { activeVisualization, indexPattern: indexId = '' } = visualization;
+  const { aggConfigParams } = activeVisualization || {};
+
+  const indexPatternsService = getIndexPatterns();
+  const indexPattern = await indexPatternsService.get(indexId);
+  const aggConfigs = getAggService().createAggConfigs(indexPattern, cloneDeep(aggConfigParams));
```

Comment:
An explanatory comment about why this is necessary might be helpful. It appears that we don't necessarily trust our `aggConfigParams` to be built with the correct `indexPattern`, but it's not clear from the code why that would be the case.

Reply:
I can add that. `AggConfigParams` does not actually have an index pattern. It's just a serialized array of aggregations and field types. We need to reconstruct `aggConfigs` which is an object using the agg service using the index pattern and serialized agg config params